virtual_rainforest/core/base_model.py

```diff
@@ -87,6 +87,10 @@ class InitialisationError(Exception):
     """Custom exception class for model initialisation failures."""
 
 
+class IntegrationError(Exception):
```

Comment:
The reason this is defined here is linked to an annoying bug/feature of `sphinx`. If you define a custom exception in a particular script then it can't be used in docstrings of functions/classes in the same script without `sphinx` producing a "reference target not found" warning.

This warning can be fixed by importing the custom exception, i.e. by moving it to be defined in another script. It's a bit annoying to have to do this in general though, as often the natural place to define a custom exception it will be a place where it's raised by one of the functions.

One possible solution is to place all custom exceptions in a `core.exceptions` module, or for each module to have a `exceptions` module of its own. If either of those things seem like sensible things to do?

Reply:
That's annoying... Maybe something to add to the next meeting on the architecture of the tool. Seems like an Sphinx bug (or at least, odd behaviour), so maybe we could raise the issue upstream.